### PR TITLE
Single-file cubemap save/load

### PR DESCRIPTION
**Pull request by guycalledfrank**

### CODE REVIEW

File: src/graphics/graphics_texture.js [hunk 1]
```diff
@@ -358,6 +358,85 @@ pc.extend(pc, function () {
...
+                    }
+                }
+                var mipSize
+                fsize += this._levels[i].length;
+                i++;
+            }
+
+            var buff = new ArrayBuffer(fsize);
+            var header = new Uint32Array(buff, 0, 128 / 4);
+
```
Comment (vkalpias) [inline src/graphics/graphics_texture.js]: Black Magic?

Comment (guycalledfrank) [inline src/graphics/graphics_texture.js]: where?

Comment (vkalpias) [inline src/graphics/graphics_texture.js]: All these numbers here ? Maybe need some comments or constants or something?

Comment (vkalpias) [inline src/graphics/graphics_texture.js]: line 397 and below..

Comment (guycalledfrank) [inline src/graphics/graphics_texture.js]: If we'll have to extend this for many different pixel formats, some things can be clarified. But for now it's enough, it's just the fastest way to write it.

File: src/resources/resources_texture.js [hunk 2]
```diff
@@ -183,32 +184,42 @@ pc.extend(pc.resources, function () {
...
-                    numBlocks = numBlocksAcross * numBlocksDown;
-                    mipSize = numBlocks * kBlockSize;
-                } else {
-                    mipSize = mipWidth * mipHeight * 4;
+            for(var face=0; face<faces; face++) {
+                var mipWidth = width;
+                var mipHeight = height;
+                for(var i=0; i<mips; i++) {
+                    if (compressed) {
+                        numBlocksAcross = Math.floor((mipWidth + kBlockWidth - 1) / kBlockWidth);
```
Comment (daredevildave) [inline src/resources/resources_texture.js]: Coding standards. Constants are ALL_CAPS.

File: src/graphics/graphics_texture.js [hunk 3]
```diff
@@ -358,6 +358,85 @@ pc.extend(pc, function () {
...
+            header[23] = 16711680;
+            header[24] = 65280;
+            header[25] = 255;
+            header[26] = 4278190080;
+            header[27] = this._levels.length===1? 4096 : 4198408;
+            header[28] = this.cubemap? 65024 : 0;
+            header[29] = 0;
+            header[30] = 0;
+            header[31] = 0;
+
```
Comment (daredevildave) [inline src/graphics/graphics_texture.js]: I would definitely make this CONST_VALUE's. When you come to extend these in 6 months time, you'll be happy you did.

**Pull request merged**